### DISCUSSION TITLE
Does Timer functionality exist in Prometheus.Client?

### DISCUSSION THREAD

**Discussion by cscoon** [PrometheusClientNet/Prometheus.Client#176] [Q&A] [answered]
I'm migrating our applications from Prometheus-net to Prometheus.Client.

I've been able to migrate all of the functionality we were using EXCEPT for the ITimer functionality.

Here is a sample of our existing code (with Prometheus-net):

```
Prometheus.Metrics.CreateHistogram(name, description,
    new HistogramConfiguration { LabelNames = MetricCollectorHelper.GetLabelNames(labels), Buckets = MetricCollectorHelper.GetHistogramBuckets(bucketType, _metricsConfiguration) })
        .WithLabels(MetricCollectorHelper.GetLabelValues(labels))
        .NewTimer()
```

Does this functionality exist in Prometheus.Client or in an extension somewhere?

**Comment by sanych-sun** [comment, marked as answer]
We have similar functionality: [ObserveDuration and ObserveDurationAsync](https://github.com/prom-client-net/prom-client/blob/13f06ec9ea27d37992cae668d78559fd7efce7dd/src/Prometheus.Client.Abstractions/ValueObserverExtensions.cs#L8) extension methods that could be used for Histogram and Summary. 

For example:

```
await myHistogram.WithLabels("a", "b").ObserveDurationAsync(async () => {
    await Task.Delay(1000);
});
```

The benefit of this approach - it's a determined moment of the observation (unlike nonintuitive at least for me observation in timer dispose).

I hope this helps =)

**Reply by sanych-sun** [reply]
@cscoon Does it answer the question? Could you please mark the response as answer? So community could rely on the response.

**Reply by cscoon** [reply]
Yes, will do!
